sale_coupon_product_management/models/sale_coupon_program_option.py

```diff
@@ -114,6 +121,8 @@ def get_program_values(self):
         for option in self:
             opt_cfg = cfg[option.option_type]
             opt_vals = opt_cfg["values"]
+            if program:  # TODO: a bit redundant.
+                option._update_values_from_program(opt_vals, program)
```

Comment:
Same comment, follow the @mmequignon proposition.
Thanks.

Reply:
Updated, but it now made implementation more complicated and harder to extend.

Also where you get this "mutable object" should never be updated inside method? Tried to look for such antipattern, but people say its not antipattern, as long as intention/method is named clearly (like mentioning `update` word or something similar).

TBF `dict.update` that you wanted to use instead of my custom method, actually does same thing. It just updates mutable object inside method/function. Looks like nitpicking to me:)

Reply:
There's plenty of examples and answers everywhere (first result https://softwareengineering.stackexchange.com/a/245809)

Regarding your comment, you're right, maybe the best place to set would be in `_get_program_values_cfg`
But there's no need to add dozen of intermediate layers where each one modify the same dict.
IMO, its the `get_program_values` to aggregate values into a single dict.

And regarding your comment in the code, it's not a bad design.

Reply:
OK, I can remove that comment if you think its a good design.

Though that link you put, just talks in general and people actually comment it depends on language and context. So its not black and white.

Also `update` function for dict just does what you say can't be done. So it contradicts saying that object can't be modified inside method/function (then you would not be able to update it at all).

https://stackoverflow.com/questions/58533878/change-object-directly-in-function-is-anti-pattern-in-python

And regarding my comment about it being bad design. It actually has nothing to do where update happens. It has to do with the fact, that now that helper method has become poorly extendable, because we would need to change returned structure and/or how returned dict is treated if any new values would need to be introduced to be used from `program` record or some other related record.

It also breaks PoS, because caller needs to know more than it should have (how to treat returned dict). So its now kind of hardcoded:)